keps/sig-cloud-provider/20180731-cloud-provider-docs.md

```diff
@@ -0,0 +1,155 @@
+      * [Non-Goals](#non-goals)
+   * [Proposal](#proposal)
+      * [Proposed Format for Documentation](#proposed-format-for-documetation)
+      * [Requirement 1: Example Manifests](#requirement-1-example-manifests)
+      * [Requirement 2: Resource Management](#requirement-2-resource-management)
+      * [Requirement 3: Kubeadm Setup Guide](#requirement-3-kubeadm-setup-guide)
+   * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+   * [Risks and Mitigations](#risks-and-mitigations)
+   * [Graduation Criteria](#graduation-criteria)
+   * [Alternatives [optional]](#alternatives-optional)
+* [Implementation History](#implementation-history)
+
+### Summary
+This KEP describes the documentation requirements for both in-tree and out-of-tree cloud providers.
+These requirements are meant to ensure the usage and integration of cloud providers with the rest of the Kubernetes ecosystem is concise and clear and that documentation layout across providers is consistent.
+
+### Motivation
+Currently documentation for cloud providers for both in-tree and out-of-tree managers is limited in both scope, consistency, and quality. This KEP describes requirements to create and maintain consistent documentation across all cloud providers. By establishing these standards, SIG Cloud Provider will benefit the user-community by offering a single discoverable source of reliable documentation while relieving the SIG-Docs team from the burden of maintaining content from various cloud providers across many Kubernetes versions.
+
+#### Goals
+
+* Create a single source of truth that outlines what documentation from each provider should cover.
+* Ensure all Kubernetes cloud providers adhere to this doc format to provide a consistent experience for all users.
+* Ensure SIG Docs can confidently link to documentation by any Kubernetes cloud provider on any future releases.
+
+#### Non-Goals
+
+* Where in the Kubernetes website the cloud provider documentation will be hosted. This should be a decision made by SIG Docs based on the content given by each cloud provider.
+* Cloud provider documentation outside the scope of Kubernetes.
+
+### Proposal
+
+#### Proposed Format for Documentation
+
+The following is a proposed format of how docs should be added to `k8s.io/cloud-provider-*` repositories.
+
+#### Requirement 1: Example Manifests
+
+Provide validated manifests for every component required for both in-tree and out-of-tree versions of your cloud provider. The contents of the manifest should contain a DaemonSet resource that runs these components on the control plane nodes or a systemd unit file. The goal of the example manifests is to provide enough details on how each component in the cluster should be configured. This should provide enough context for users to build their own manifests if desired.
+
+```
+cloud-provider-foobar/
+├── ...
+├── ...
+├── docs
+│   └── example-manifests
+│       └── in-tree/
```

Comment:
I would include a `configuration.manifest` as configuration is shared amongst the various deployment styles, it also requires the most annotations (see previous comment)

Reply:
I might be missing something here, what would `configuration.manifest` contain?

Reply:
a configuration example for the cloud provider, something like this: https://github.com/kubernetes/cloud-provider-vsphere/blob/master/docs/deploying_cloud_provider_vsphere_with_rbac.md#3-optional-but-recommended-storing-vcenter-credentials-in-a-kubernetes-secret

Reply:
Ahh I see, how about `configuration.md` file then? I think just showing an example manifest here wouldn't be enough cause there's some contexting missing around specific provider details? The `*.manifest` file should ideally be just a DaemonSet spec or systemd unit file that can be referenced as an example for common components. (I'll add more details on this above). Thoughts?